File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
    l = 0
    (1...nums.size).each do |r|
        if nums[l] != nums[r]
            l += 1
            nums[l], nums[r] = nums[r], nums[l]
        end
    end
    
    l + 1
end

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const removeDuplicates = function(nums) {
    let l = 0;
    
    for (let r = 0; r < nums.length; r++) {
        if (nums[r] !== nums[r + 1]) {
            nums[l] = nums[r];
            l++;
        }
    }
    
    return l;
};